app/controllers/register_controller.rb

```diff
@@ -0,0 +1,24 @@
+class RegisterController < ApplicationController
+  def new
+    @user_info = {}
+  end
+
+  def create
+    user = User.create(user_params)
+    if user.save
+      flash[:notice] = "Welcome #{params[:name]} Registration Successful"
+      session[:user_id] = user.id
+      redirect_to "/profile"
+    else
+      flash[:error] = user.errors.full_messages.to_sentence
+      @user_info = user_params
+      render :new
+    end
+  end
```

Comment:
We might want to pull out some of this logic into private helper methods to adhere to Sandi Metz' Rules For Developers.